native/src/main/java/io/ballerina/stdlib/io/nativeimpl/CsvChannelUtils.java

```diff
@@ -0,0 +1,152 @@
+        BObject textRecordChannel = ValueCreator.createObjectValue(getIOPackage(),
+            READABLE_TEXT_RECORD_CHANNEL, characterChannel, FIELD_SEPERATOR, ROW_SEPERATOR, FORMAT);
+        BObject recordIterator = ValueCreator.createObjectValue(getIOPackage(), CSV_ITERATOR);
+        recordIterator.addNativeData(CSV_RETURN_TYPE, typeDesc);
+        recordIterator.addNativeData(ITERATOR_NAME, textRecordChannel);
+        return ValueCreator.createStreamValue(
+                TypeCreator.createStreamType(describingType), recordIterator);
+    }
+
+    public static Map<String, Object> getStruct(String[] fields, final StructureType structType) {
+        Map<String, Field> internalStructFields = structType.getFields();
+        int fieldLength = internalStructFields.size();
+        Map<String, Object> struct = null;
+        if (fields.length > 0) {
+            Iterator<Map.Entry<String, Field>> itr = internalStructFields.entrySet().iterator();
+            struct = new HashMap<>();
+            for (int i = 0; i < fieldLength; i++) {
+                final Field internalStructField = itr.next().getValue();
+                int type = TypeUtils.getReferredType(internalStructField.getFieldType()).getTag();
+                String fieldName = internalStructField.getFieldName();
+                if (fields.length > i) {
+                    String value = fields[i];
+                    if (value == null || value.isEmpty()) {
+                        if (type == TypeTags.UNION_TAG) {
+                            List<Type> members = ((UnionType) internalStructField.getFieldType()).getMemberTypes();
+                            if (TypeUtils.getReferredType(members.get(1)).getTag() == TypeTags.NULL_TAG) {
+                                struct.put(fieldName, null);
+                                continue;
+                            } 
+                            throw IOUtils.createError("Unsupported nillable field : " + fieldName);
+                        }
+                        throw IOUtils.createError("Following field does not support nill values: " + fieldName);
+                    } 
+                    if (type == TypeTags.UNION_TAG) {
+                        List<Type> members = ((UnionType) internalStructField.getFieldType()).getMemberTypes();
+                        if (TypeUtils.getReferredType(members.get(1)).getTag() == TypeTags.NULL_TAG) {
+                            type = TypeUtils.getReferredType(members.get(0)).getTag();
+                        } else {
+                            throw IOUtils.createError("Unsupported nillable field : " 
+                                + fieldName + " for value: " + value);
+                        }
+                    }
+                    String trimmedValue = value.trim();       
+                    switch (type) {
+                        case TypeTags.INT_TAG:
+                            struct.put(fieldName, Long.parseLong(trimmedValue));
```

Comment:
What if there are values which cannot be parsed to int? What error is thrown here?